koku/sources/api/source_status.py

```diff
@@ -185,4 +174,10 @@ def source_status(request):
         # Source isn't in our database, return 404.
         return Response(status=status.HTTP_404_NOT_FOUND)
 
-    return _deliver_status(request, source_status_obj)
+    # Keeping prior functionality if flag is disabled
+    if not is_status_api_update_enabled(source_status_obj.source.account_id):
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    status_data = source_status_obj.status()
```

Comment:
I might be following along wrong so sorry if thats the case. But should this actually be calling `sources_response` to build the source status rather then sending a `None` or status `error_obj` which is returned by `status`?